### PR TITLE
create/run: add ipc flag to create and run commands

### PR DESCRIPTION
**Pull request by 89luca89**
Right now nerdctl inspect does not have a --ipc flag similar to
docker and podman.

This PR aims to add this, allowing for the basic values:

    private
    host

It will default to private to match docker's behaviour

Example of the command:

```sh
./nerdctl create --ipc host --name test alpine:latest sleep infinity
./nerdctl start test
./nerdctl exec -ti test ls -la /dev
```

output:

```console
drwxrwxrwt    2 nobody   nobody          40 May 12 09:35 /dev/mqueue
drwxrwxrwt    2 nobody   nobody          80 May 12 10:35 /dev/shm
```

Run example:

```sh
./nerdctl run --ipc host --name test --rm -ti alpine:latest ls -lad /dev/{mqueue,shm}
```

output:

```console
drwxrwxrwt    2 nobody   nobody          40 May 12 09:35 /dev/mqueue
drwxrwxrwt    2 nobody   nobody          80 May 12 10:35 /dev/shm
```

Signed-off-by: Luca Di Maio <luca.dimaio1@gmail.com>

### CODE REVIEW

**Comment by 89luca89** [comment]
Hi @AkihiroSuda 
Not sure why the integration test is failing, as I didn't change default behavior 
Seems erroring at installing containerd-rootless.sh so probably just a fluke

File: cmd/nerdctl/run.go [hunk 1]
```diff
@@ -138,6 +138,7 @@ func setCreateFlags(cmd *cobra.Command) {
 	cmd.Flags().StringP("hostname", "h", "", "Container host name")
 	// #endregion
 
+	cmd.Flags().String("ipc", "", "IPC namespace to use")
```
Comment (AkihiroSuda) [inline cmd/nerdctl/run.go:141]: Could you add shell completion?  Just like `cmd.RegisterFlagCompletionFunc("cgroupns", ...)` in L153-154

File: cmd/nerdctl/run_linux.go [hunk 2]
```diff
@@ -175,5 +175,17 @@ func setPlatformOptions(opts []oci.SpecOpts, cmd *cobra.Command, id string) ([]o
...
+	// if nothing is specified, or if private, default to normal behavior
+	if ipc == "host" {
+		opts = append(opts, oci.WithHostNamespace(specs.IPCNamespace))
+		opts = append(opts, withBindMountHostIPC)
+	} else if ipc != "" && ipc != "private" {
+		return nil, fmt.Errorf("error: %v", "invalid ipc value, supported values are 'private' or 'host'")
+	}
+
 	return opts, nil
 }
```
Comment (AkihiroSuda) [inline cmd/nerdctl/run_linux.go:191]: Can we have an integration test in `run_linux_test.go`?

File: cmd/nerdctl/run.go (hunk 1 above)
Comment (89luca89) [reply cmd/nerdctl/run.go:141]: Done! Sorry completely missed this :+1:

File: cmd/nerdctl/run_linux.go (hunk 2 above)
Comment (89luca89) [reply cmd/nerdctl/run_linux.go:191]: Done! :+1:

**Comment by 89luca89** [comment]
Hi @AkihiroSuda cannot see what is failing on `Linux/cgroup2` test, but in the other integration tests seems like `TestRunIpcHost` is successful

**Comment by AkihiroSuda** [comment]
Added a follow-up commit: https://github.com/containerd/nerdctl/pull/1057/commits/f66b94375a87f05b1245aa493c2d1ca36730dce3

**Comment by 89luca89** [comment]
> Added a follow-up commit: [f66b943](https://github.com/containerd/nerdctl/commit/f66b94375a87f05b1245aa493c2d1ca36730dce3)

Oh yea much better this way, thanks 😄

**Pull request merged**